eth/downloader/queue.go

```diff
@@ -319,49 +483,86 @@ func (q *queue) Reserve61(p *peer, count int) *fetchRequest {
 		Hashes: send,
 		Time:   time.Now(),
 	}
-	q.pendPool[p.id] = request
+	pendPool[p.id] = request
 
 	return request
 }
 
-// Reserve reserves a set of headers for the given peer, skipping any previously
-// failed download. Beside the next batch of needed fetches, it also returns a
-// flag whether empty blocks were queued requiring processing.
-func (q *queue) Reserve(p *peer, count int) (*fetchRequest, bool, error) {
+// ReserveBodies reserves a set of body fetches for the given peer, skipping any
+// previously failed downloads. Beside the next batch of needed fetches, it also
+// returns a flag whether empty blocks were queued requiring processing.
+func (q *queue) ReserveBodies(p *peer, count int) (*fetchRequest, bool, error) {
+	noop := func(header *types.Header) bool {
```

Comment:
Perhaps call it `isNoop` since it does something and is used to see whether *another thing* will be a noop